### PR TITLE
Us859 - As a developer I want to refactor the rules manager folder to more easily integrate it into the game flow

### PR DESCRIPTION
**Pull request by tsaldrid**
Tasks 860 through 867, 895.

Some of the proposed changes (removing non-permanent cost modifiers, and refactoring cost modifier service to be a singleton) ended up being beyond the scope of this integration audit, as they require a bit more research and planning on how to accomplish them for when their use is implemented as features. 

There are some failing unit tests, however none appear to be caused by these changes.

### CODE REVIEW

File: Assets/Scripts/Managers/Purchase/PurchaseManager.cs [hunk 1]
```diff
@@ -13,7 +13,7 @@ public class PurchaseManager : MonoBehaviour
     {
         [Header("Dependencies")]
         [SerializeField] private OwnershipServiceAdapter ownership;
-        [SerializeField] private IRuleSet rules = new StandardRuleSet();
+        [SerializeField] private StandardRuleSet rules = StandardRuleSet.GetInstance();
```
Comment (hdathert) [inline Assets/Scripts/Managers/Purchase/PurchaseManager.cs]: If we're resolving a reference to a singleton, we don't need it to be serialized. Technically, we don't even need to cache it, but it does make later code more readable.

File: Assets/Scripts/Managers/Rules/StandardRuleSet.cs [hunk 2]
```diff
@@ -8,38 +9,47 @@ namespace Assets.Scripts.Managers.Rules
     /// Will be populated when rent rule logic is brought oever
     /// </summary>
     [System.Serializable]
-    public class StandardRuleSet : IRuleSet
+    public class StandardRuleSet
```
Comment (hdathert) [inline Assets/Scripts/Managers/Rules/StandardRuleSet.cs:12]: I'm wondering if this would be better as a `ScriptableObject` and moved into our `Data/` folder to keep strictly data classes together.

Comment (hdathert) [reply Assets/Scripts/Managers/Rules/StandardRuleSet.cs:12]: If we did that, we should probably rename it to `ConfigurableRules` or something, since that is more accurate to what it is. The default values can all remain- we can spawn it with them.

File: Assets/Scripts/Managers/Purchase/PurchaseManager.cs (hunk 1 above)
Comment (hdathert) [inline Assets/Scripts/Managers/Purchase/PurchaseManager.cs]: If we're getting a reference to a singleton, then we don't need the `[SerializeField]` attribute.

File: Assets/Scripts/Managers/Rules/StandardRuleSet.cs (hunk 2 above)
Comment (hdathert) [inline Assets/Scripts/Managers/Rules/StandardRuleSet.cs:12]: It seems like we should be making this a ScriptableObject at some point soon, so that we can asset-ify it and create preset configurations. When we do that, we should probably also move this to the `Data/` folder with the other data container classes, its out of place in the `Managers/` folder.

Additionally, it seems like it would be better named as `RulesConfiguration` or something.

File: Assets/Scripts/Managers/PlayerManager.cs [hunk 3]
```diff
@@ -305,12 +303,7 @@ public void OnPayPlayerEvent(PayPlayerEvent payPlayerEvent)
...
     private void EnsureDependencies()
     {
-        if (!rulesManager)
-            rulesManager = FindFirstObjectByType<RulesManager>();
-
         if (activeRuleset == null)
-            activeRuleset = rulesManager != null
-                ? rulesManager.ActiveRules
-                : new StandardRuleSet();
+            activeRuleset = StandardRuleSet.GetInstance();
```
Comment (hdathert) [inline Assets/Scripts/Managers/PlayerManager.cs]: This will be resolved in the merge- `EnsureDependancies` was removed from `PlayerManager` in another story- the rules that were used are just in-line constants for now, so the merge should be super straight-forward, replacing those constants with a static call to the current rules for Starting Money.

**Review by hdathert — CHANGES_REQUESTED** [review]
Needs a few small changes outlined below. The change into a scriptable object should be a future story.

The merge conflict should be straightforward, one of my comments outlines how that should be resolved. I would suggest doing that in an IDE instead of the github editor.

I'd like to see a story created about the other parts of the audit that were "out of scope", since it was not elaborated at all in the description of this PR. Just because we're deferring them, doesn't mean we can't document what we're missing.

**Review by hdathert — APPROVED** [review]
# Code Review Checklist

**Completed for**: US #859 and Tasks #860 to #896  
**Code Review Completed By**: @hdathert 

Use this checklist to review new code changes before merging.

---

## Implementation
- [x] Does this code change accomplish what it is supposed to do?
- [x] Can the solution be simplified?
- [x] This change does **not** add unwanted compile-time or run-time dependencies.
- [x] The code is at the right abstraction level.
- [x] The code is modular enough.
- [x] The solution is adequate for maintainability, readability, performance, and security.
- [x] Similar functionality does **not** already exist in the codebase (or reuse is not appropriate).
- [x] No additional best practices or design patterns would substantially improve this code.

---

## Logic Errors and Bugs
- [x] No use cases where the code behaves incorrectly.
- [x] No inputs or external events that can break the code.
- [x] Code agrees with specifications.
- [x] No off-by-one errors.
- [x] No magic numbers (or they are well-named constants).
- [x] Proper use of global/large-scope variables (minimized and justified).

---

## Error Handling and Logging
- [x] Error handling is implemented correctly.
- [x] Error messages are user-friendly (where applicable).
- [x] Sufficient, meaningful log events exist for debugging (no sensitive data in logs).

---

## Dependencies
- [x] Documentation/configuration/README updates were made as needed.
- [x] No negative impact on other parts of the system or backward compatibility.

---

## Security and Data Privacy
- [x] No new security vulnerabilities introduced.
- [x] All user input is validated, sanitized, and escaped (XSS/SQL injection prevented).
- [x] Data from external APIs/libraries is handled safely and checked as needed.

---

## Performance
- [x] No significant performance regressions.
- [x] No obvious opportunities for significant performance gains left unaddressed.

---

## Ethics and Morality
- [x] Appropriate measures exist to prevent or report harassment/abuse (if applicable).

---

## Testing and Testability
- [x] Code is testable.
- [x] Automated tests have been added or updated.
- [x] Existing tests sufficiently cover the change (unit/integration/system).
- [x] Edge cases and boundary inputs are tested.

---

## Readability
- [x] Code is easy to understand.
- [x] Methods/functions are appropriately small and focused.
- [x] Names (functions/classes/variables) are clear and descriptive.
- [x] Code is in the correct file/folder/package.
- [x] Control flow is intuitive; data flow is clear.
- [x] Comments are clear, relevant, and up to date.
- [x] No commented-out or redundant code remains.
- [x] No overly dense one-liners; no overly long methods.

---

**Use this before submitting a pull request.**  
It helps ensure code quality, maintainability, and compliance with best practices.

*Based on awesome code review guides and MIT’s code review recommendations.*

**Comment by dzadroga** [comment]
# Code Review Checklist

**Completed for**: US #859 and Tasks #860 #867 and #895
**Code Review Completed By**: Dakota Zadroga

Use this checklist to review new code changes before merging.

---

## Implementation
- [X] Does this code change accomplish what it is supposed to do?
- [X] Can the solution be simplified?
- [X] This change does **not** add unwanted compile-time or run-time dependencies.
- [X] The code is at the right abstraction level.
- [X] The code is modular enough.
- [X] The solution is adequate for maintainability, readability, performance, and security.
- [X] Similar functionality does **not** already exist in the codebase (or reuse is not appropriate).
- [X] No additional best practices or design patterns would substantially improve this code.

---

## Logic Errors and Bugs
- [X] No use cases where the code behaves incorrectly.
- [X] No inputs or external events that can break the code.
- [X] Code agrees with specifications.
- [X] No off-by-one errors.
- [X] No magic numbers (or they are well-named constants).
- [X] Proper use of global/large-scope variables (minimized and justified).

---

## Error Handling and Logging
- [X] Error handling is implemented correctly.
- [X] Error messages are user-friendly (where applicable).
- [X] Sufficient, meaningful log events exist for debugging (no sensitive data in logs).

---

## Dependencies
- [X] Documentation/configuration/README updates were made as needed.
- [X] No negative impact on other parts of the system or backward compatibility.

---

## Security and Data Privacy
- [X] No new security vulnerabilities introduced.
- [X] All user input is validated, sanitized, and escaped (XSS/SQL injection prevented).
- [X] Data from external APIs/libraries is handled safely and checked as needed.

---

## Performance
- [X] No significant performance regressions.
- [X] No obvious opportunities for significant performance gains left unaddressed.

---

## Ethics and Morality
- [X] Appropriate measures exist to prevent or report harassment/abuse (if applicable).

---

## Testing and Testability
- [X] Code is testable.
- [X] Automated tests have been added or updated.
- [X] Existing tests sufficiently cover the change (unit/integration/system).
- [X] Edge cases and boundary inputs are tested.

---

## Readability
- [X] Code is easy to understand.
- [X] Methods/functions are appropriately small and focused.
- [X] Names (functions/classes/variables) are clear and descriptive.
- [X] Code is in the correct file/folder/package.
- [X] Control flow is intuitive; data flow is clear.
- [X] Comments are clear, relevant, and up to date.
- [X] No commented-out or redundant code remains.
- [X] No overly dense one-liners; no overly long methods.

---

**Use this before submitting a pull request.**  
It helps ensure code quality, maintainability, and compliance with best practices.

*Based on awesome code review guides and MIT’s code review recommendations.*
